src/arrays/constructors.jl

```diff
@@ -16,4 +16,22 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+# Reference :
+# Section - 1.1, http://cds.cern.ch/record/331607/files/9708012.pdf
+function coherentstatevec_inf(n::Int,alpha::Number)
+    s = zeros(typeof(float(alpha)),n)
+    s[1] = one(alpha)
+    for i in [2:n]
+        s[i] = alpha/sqrt(i-1)*s[i-1]
+    end
+    z = QuArray(s)
+    return scale!(exp(-abs2(alpha)/2),z)
+end
+
+# Reference :
+# 2nd Defintion : http://en.wikipedia.org/wiki/Coherent_states#Mathematical_features_of_the_canonical_coherent_states
+coherentstatevec(n::Int, alpha::Number) = displaceop(n,alpha)*statevec(1,FiniteBasis(n))
+
+export statevec,
+    coherentstatevec,
+    coherentstatevec_inf
```

Comment:
You can call it via `QuBase.coherentstatevec_inf`. (In Julia you can always access all methods and types defined in a module by putting `Modulename.` in front. Exported methods are in a way just a convenience and in that sense define a "public API".)